iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -1175,6 +1177,11 @@ internal string PopulateMessagePropertiesFromMessage(string topicName, Message m
                 msg = topicName;
             }
 
+            if (Encoding.ASCII.GetByteCount(msg) > MaxTopicNameLength)
```

Comment:
Why do you need the encoding? The `msg` is passed as-is to message. I doubt ASCII is the right char-set - likely a small subset anyway. 
I think you can assume correct chars are used and simply count characters instead of total bytes from the string.